contracts/token/ERC20/EncryptedERC20.sol

```diff
@@ -1,201 +1,182 @@
-        return balances[msg.sender];
+    /**
+     * @notice See {IEncryptedERC20-allowance}.
+     */
+    function allowance(address owner, address spender) public view virtual returns (euint64) {
+        return _allowance(owner, spender);
     }
 
-    // Sets the `encryptedAmount` as the allowance of `spender` over the caller's tokens.
-    function approve(address spender, bytes calldata encryptedAmount) public virtual returns (bool) {
-        approve(spender, TFHE.asEuint64(encryptedAmount));
-        return true;
+    /**
+     * @notice See {IEncryptedERC20-balanceOf}.
+     */
+    function balanceOf(address account) public view virtual returns (euint64) {
+        return _balances[account];
     }
 
-    // Sets the `amount` as the allowance of `spender` over the caller's tokens.
-    function approve(address spender, euint64 amount) public virtual returns (bool) {
-        address owner = msg.sender;
-        _approve(owner, spender, amount);
-        emit Approval(owner, spender);
-        return true;
+    /**
+     * @notice See {IEncryptedERC20-name}.
+     */
+    function name() public view virtual returns (string memory) {
+        return _name;
     }
 
-    // Returns the remaining number of tokens that `spender` is allowed to spend
-    // on behalf of the `owner`. The returned ciphertext is under the caller's `publicKey`.
-    function allowance(
-        address owner,
-        address spender,
-        bytes32 publicKey,
-        bytes calldata signature
-    ) public view virtual onlySignedPublicKey(publicKey, signature) returns (bytes memory) {
-        require(owner == msg.sender || spender == msg.sender, "Caller must be owner or spender");
-        return TFHE.reencrypt(_allowance(owner, spender), publicKey);
+    /**
+     * @notice See {IEncryptedERC20-symbol}.
+     */
+    function symbol() public view virtual returns (string memory) {
+        return _symbol;
     }
 
-    // Transfers `encryptedAmount` tokens using the caller's allowance.
-    function transferFrom(address from, address to, bytes calldata encryptedAmount) public virtual returns (bool) {
-        transferFrom(from, to, TFHE.asEuint64(encryptedAmount));
-        return true;
+    /**
+     * @notice See {IEncryptedERC20-totalSupply}.
+     */
+    function totalSupply() public view virtual returns (uint64) {
+        return _totalSupply;
     }
 
-    // Transfers `amount` tokens using the caller's allowance.
-    function transferFrom(address from, address to, euint64 amount) public virtual returns (bool) {
-        address spender = msg.sender;
-        (ebool isTransferable, euint8 errorCode) = _updateAllowance(from, spender, amount);
-        _transfer(from, to, amount, isTransferable, errorCode);
-        return true;
+    function _approve(address owner, address spender, euint64 amount) internal virtual {
+        _allowances[owner][spender] = amount;
+        TFHE.allow(amount, address(this));
```

Comment:
Please replace all occurences of `TFHE.allow(handle,address(this))` by the new `TFHE.allowThis(handle)`